### PR TITLE
Fixes #29 - make awesome_print optional

### LINKED ISSUE #29: Don't explicitly depend on and require awesome_print, let users opt-in
Currently, apipie bindings has a runtime dependency on  `awesome_print` in the gemspec and requires it in api.rb[1]

AwesomePrint is a great debugging for getting started with `apipie-bindings` but should be avoided in production because it's not really meant for production.  It globally monkey-patches core and library classes on require which can easily break your application when upgrading those classes or have performance degradations[2].  

If awesome_print is really needed or required, a developer can modify their application to use it with `apipie-bindings` by adding  `gem, 'awesome_print'` to the Gemfile without a require value, so it gets required.

And in `apipie-bindings`, we should remove the explicit dependency in the gemspec so users can opt-in explicitly.

Then, in api.rb we can do something like `object.respond_to?(:ai) ? object.ai : object.inspect`
Or, we could have a method that checks for the AwesomePrint constant to see if it's loaded so we don't have to use these ternaries all over the place.

Please let me know if I can help do this or review this work.  I'm sure there's better ways of doing this, so feel free to do this differently.

Thanks in advance :sunglasses: 

#### [1] Calls to .ai (awesome_inspect)
```
$ git grep -E "\bap\b|\bai\b|\bawesome|\bAwesome"
apipie-bindings.gemspec:  s.add_dependency 'awesome_print'
lib/apipie_bindings/api.rb:require 'awesome_print'
lib/apipie_bindings/api.rb:      log.debug "Global headers: #{headers.ai}"
lib/apipie_bindings/api.rb:      log.debug "Params: #{params.ai}"
lib/apipie_bindings/api.rb:      log.debug "Headers: #{headers.ai}"
lib/apipie_bindings/api.rb:            (e.respond_to?(:response) ? process_data(e.response).ai : e.ai)
lib/apipie_bindings/api.rb:      log.debug "Response: %s" % (options[:reduce_response_log] ? "Received OK" : result.ai)
lib/apipie_bindings/api.rb:      log.debug "Response headers: #{response.headers.ai}" if response.respond_to?(:headers)
```

#### [2]  Benchmark of Array#grep from stdlib on 2.2.3
**Note** If your array is larger or passes a block to `grep`, I'm sure the performance of the monkey patch is even worse.

```
$ ruby -e "require 'benchmark/ips'; Benchmark.ips { |x| x.report('Regular Array#grep') { methods.grep(/get/) } }"
Calculating -------------------------------------
  Regular Array#grep     1.908k i/100ms
-------------------------------------------------
  Regular Array#grep     20.271k (± 4.1%) i/s -    103.032k
```

#### Benchmark of Array#grep monkey patched by awesome_print

```
$ ruby -e "require 'benchmark/ips'; require 'awesome_print'; Benchmark.ips { |x| x.report('Array#grep with AwesomePrint\'s patch') { methods.grep(/get/) } }"
Calculating -------------------------------------
Array#grep with AwesomePrint's patch
                         1.696k i/100ms
-------------------------------------------------
Array#grep with AwesomePrint's patch
                         17.082k (± 3.2%) i/s -     86.496k
```

### PR DESCRIPTION
**Pull request by mbacovsky**
Logging use awesome_print to inspect data only if it is available.

### CODE REVIEW

File: lib/apipie_bindings/api.rb [hunk 1]
```diff
@@ -1,7 +1,6 @@
 require 'json'
 require 'rest_client'
 require 'oauth'
-require 'awesome_print'
```
Comment (jrafanie) [inline lib/apipie_bindings/api.rb]: :tada:

File: apipie-bindings.gemspec [hunk 2]
```diff
@@ -26,7 +26,6 @@ EOF
   s.add_dependency 'json', '>= 1.2.1'
   s.add_dependency 'rest-client', '>= 1.6.5'      # lower versions don't allow setting infinite timeouts
   s.add_dependency 'oauth'
-  s.add_dependency 'awesome_print'
```
Comment (jrafanie) [inline apipie-bindings.gemspec]: :heart:

**Comment by jrafanie** [comment]
@mbacovsky Looks good to me.  Thanks!!!

You might want to update the README with `Add gem "awesome_print" to your application Gemfile if you want to debug api calls with awesome inspect`

**Comment by mbacovsky** [comment]
Thanks for review, @jrafanie . README updated with your suggestion. 
Expect new release soon :wink:

**Comment by kbrock** [comment]
thanks @mbacovsky

**Pull request merged**